onmt/translate/translator.py

```diff
@@ -382,17 +400,23 @@ def translate(
                         srcs = trans.src_raw
                     else:
                         srcs = [str(item) for item in range(len(attns[0]))]
-                    header_format = "{:>10.10} " + "{:>10.7} " * len(srcs)
-                    row_format = "{:>10.10} " + "{:>10.7f} " * len(srcs)
-                    output = header_format.format("", *srcs) + '\n'
-                    for word, row in zip(preds, attns):
-                        max_index = row.index(max(row))
-                        row_format = row_format.replace(
-                            "{:>10.7f} ", "{:*>10.7f} ", max_index + 1)
-                        row_format = row_format.replace(
-                            "{:*>10.7f} ", "{:>10.7f} ", max_index)
-                        output += row_format.format(word, *row) + '\n'
-                        row_format = "{:>10.10} " + "{:>10.7f} " * len(srcs)
+                    output = report_matrix(srcs, preds, attns)
+                    if self.logger:
```

Comment:
Are there some cases where we don't have a logger?

Reply:
Exactly! I just verified that the logger has not correctly initialized. As in both call of `build_translator` in `translate.py` and `translation_server.py`, the argument logger has not been passed in. Therefore, we actually only call os.write. The same for the method self._log.
Maybe we could create another separate PR to fix them?